source/server/spongineer/debugging.rst

```diff
@@ -119,3 +119,102 @@ A ``NoClassDefFoundError`` occurs when the plugin tries to access a class that i
+    /* Minecraft Forge : 14.23.4.2705                                                                                             */
+    /******************************************************************************************************************************/
+
+This stacktrace contains the most important version numbers, as well some information about the phase the server was in.
+In this case a ``NullPointerException`` has been thrown during ``EntityTickPhase``. At this point its important to check
+which plugins are involved with the crash. This requires some research as you have to match the package names with mod
+names (Also check the ``Caused by`` blocks).
+
+* ``java`` classes can be ignored during the error search.
+* ``net.minecraft`` is the vanilla Minecraft code. If only these elements are present, its either a Minecraft bug or a
+  really nasty one (I explain those later).
+* ``org.spongepowered`` is from Sponge itself, having only these and Minecraft packages present usually indicates a
+  Sponge bug (or a nasty one). The :doc:`bugreport` chapter explains where to report bugs.
+* Other classes have to be mapped to their mods by hand. In this case there is an entry ``com.example.extendedaiplugin`` 
+  which gives us the following hints. ``com`` probably from an English speaking mod author. This could also be a
+  language code such as ``de`` or ``zh``. The next part usually indicates the mod authors name, but in some cases it may
+  also point to a hosting platform such as ``github``. In those cases you should also take a look at the next parts.
+  After that there is usually something related to the mod itself. ``extendedaiplugin`` might be related to the plugin
+  name. Please be aware that the mod may use additional separator characters such as ``-`` or spaces or use a different
+  character case for some parts. However this strategy is not always accurate, even less so in mod packs or bundles.
+
+.. note::
+
+    If you encounter a bug its usually a good idea to create a backup first, then trying to reproduce it, then narrowing
+    it down by removing mods. Only then should you report the error. If the error occurs while only non Sponge mods are
+    present try removing Sponge. If the error persists its not related to Sponge. Its usually a good idea to report bugs
+    to the mod authors first as they have good knowledge of the parts of code they are working with. However you can
+    always contact us through IRC or other means. Please provide logs for bug reports, if possible.
+
+Nasty bugs: Minecraft modding uses some advanced technics such as Mixins and ClassLoaderTransformations, which means
+that although a Minecraft class has been reported as the cause, it does not mean the code executed inside is from
+Minecraft itself. Sponge and other plugins hook into the native methods and execute their own code parts; such as
```

Comment:
code parts -> "code elements" (or components, or just "code")

Reply:
Did this get missed? Also, the semicolon needs to be a comma.

Reply:
AFAICT not + Fixed